pl_bolts/loggers/azureml.py

```diff
@@ -0,0 +1,87 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Azure Machine Learning
+----------------------
+"""
+
+
+import uuid
+from typing import Optional
+
+from pytorch_lightning.loggers import MLFlowLogger
+
+try:
+    from azureml.core import Run as AzureMlRun
+except ImportError:  # pragma: no-cover
+    AzureMlOfflineRun = None
+    _AZURE_ML_AVAILABLE = False
+else:
+    _AZURE_ML_AVAILABLE = True
+
+
+class AzureMlLogger(MLFlowLogger):
+    r"""
+    Log using `Azure Machine Learning <https://docs.microsoft.com/en-us/azure/machine-learning/>`_.
+    Install it with pip:
+
+    .. code-block:: bash
+
+        pip install mlflow azureml-mlflow
+
+    The Azure Machine Learning logger will log to standard output if running in
+    offline mode, or to
+    `Azure Machine Learning metrics <https://docs.microsoft.com/en-us/azure/machine-learning/how-to-track-experiments>`_
+    via
+    `MLFlow <https://docs.microsoft.com/en-us/azure/machine-learning/how-to-use-mlflow>`_
+    if running remotely.
+
+    **Online and offline mode**
+
+    Example::
+
+        from azureml.core import Run
+        from pytorch_lightning.loggers import AzureMlLogger
+
+        # Optional: this is the default value if no run argument is provided.
+        run = Run.get_context()
+        azureml_logger = AzureMlLogger(run)
+        trainer = Trainer(max_epochs=10, logger=azureml_logger)
+
+    Args:
+        run: Optionally inject Azure Machine Learning `Run` object directly.
+            If this is not provided, default to `Run.get_context()`.
+    """
+
+    def __init__(self, run: Optional[AzureMlRun] = None):
+
+        if not _AZURE_ML_AVAILABLE:
+            raise ImportError(
+                "You want to use `azureml-sdk` logger which is not installed yet,"
+                " install it with `pip install azureml-sdk`."
+            )
+
+        if run is None:
+            run = AzureMlRun.get_context(allow_offline=True)
+
+        try:
+            experiment = run.experiment
+            tracking_uri = experiment.workspace.get_mlflow_tracking_uri()
+            experiment_name = experiment.name
+        except AttributeError:
+            tracking_uri = None
+            experiment_name = str(uuid.uuid4())
+
+        super().__init__(experiment_name, tracking_uri)
```

Comment:
FYI, I don't think this is right for the case where you already have an AzureML run id. MLFlowLogger will synthesize a new run id and ignore the old one, causing you to end up with metrics etc logged to what AzureML thinks is a new run than the one it spawned (and where stdout is being logged to). I think the right thing is to thread through rum.run_id to MLFlow, but that's looking a bit non-trivial because of the way the MLFlow logger is set up.

Reply:
Here's what I mean (censored picture). The bottom row is the run I started, the row above it is the one that mlflow is logging to. If you start a bunch of runs at the same time it's hard to tell what goes with what.

![image](https://user-images.githubusercontent.com/9633/97120600-80d1a480-16d5-11eb-85eb-19237f1ebaf0.png)